cart/player.go

```diff
@@ -128,7 +128,12 @@ func (player *Player) HandleInteraction(t int32) {
 	player.Move_fx.Stop()
 }
 
-func (player *Player) Update(t int32, world *World, game *Game, desired *RetrievableItem) {
+func (player *Player) Update(t int32, world *World, game *Game, desired *RetrievableItem, rabbit *Rabbit) {
+	if rabbit.IsDead() && rabbit.PointInZone(player.X, player.Y) {
+		game.ChangeState(GAME_STATE_OVER)
+		player.SetDead()
+	}
```

Comment:
Touching the dead rabbit kills you, but the rabbit dying is OK?

Actually, I think this makes sense.

Reply:
You should return immediately after dying. Otherwise the remainder of the update could put the player into a weird state.

Reply:
Yes true, now that we have a `SetDead` function we don't have to wait for the call to animate to update the sprite.